### ISSUE TITLE
Angular 5.0.5 doesn't work with RxJS 5.5.3

### ISSUE THREAD

**Issue by Cito** [angular/angular#20752]
## I'm submitting a...

<pre><code>
[X] Bug report  
</code></pre>

## Current behavior
When RxJS 5.5.3 ist used, I'm getting an "EmptyError: no elements in sequence" when trying to navigate using the Angular Router.

## Expected behavior
Should navigate without errors, like with RxJS 5.5.2.

## Minimal reproduction of the problem with instructions
Run [this Plunk](https://plnkr.co/mBVUqYWouYnlpD8mmXA0). You can navigate between 2 pages without problems. Then edit the config.js file to make it load rxjs 5.5.3 instead of 5.5.2. Run the Plunk again. Now when you click on any of the two links, you get the above error in the console, and the navigation does not happen.

## Environment

<pre><code>
Angular version: 5.0.5

Browser:
- [X] Chrome (desktop)
- [X] Firefox
- [X] Edge
</code></pre>

## Remarks

* See also [RxJS issue #3151](https://github.com/ReactiveX/rxjs/issues/3151)
* The problem seems to be triggered by routes with an empty path

**Comment by kaaterskil** [comment]
Seeing the same behavior with children and empty child paths. Reverting to 5.5.2 resolves the problem.

**Comment by fabiogodoy** [comment]
Hello everyone I face with the same error, but I'd found a workaround, the problem only happens when we have an empty path, so adding pathMatch: 'full' works for me:
{
    path: '',
    component: HomeComponent,
    pathMatch: 'full'
  },
  {
    path: 'about',
    component: AboutComponent,
    pathMatch: 'full'
  }

**Comment by fergalmoran** [comment]
The fix @kaaterskil posted above works for me also, huge thanks!!

**Comment by AbhishekKaura25** [comment]
Reverting back to 5.5.2 worked for me also

**Comment by ericmartinezr** [comment]
They already submitted a fix https://github.com/ReactiveX/rxjs/issues/3151#issuecomment-348730503. This issue is not angular's.

**Comment by Cito** [comment]
@ericmartinezr They just reverted another fix in 5.5.3 that caused the problem. The real cause of the issue must still be investigated and might reveal an issue in how Angular is using rxjs.

**Comment by MickL** [comment]
I reverted to 5.5.2 which seem to work

**Comment by Stradivario** [comment]
With 5.5.2 everything worked as expected!
Best Regards guys ! Thank you!

**Comment by ocombe** [comment]
Hello, I'm closing this since it is a bug in RXJS and not Angular.

**Comment by uloga** [comment]
Have the same issue with RXJS 5.5.2 @angular 5.0.5  when using child routes with wildcard for 404 NotFound
it works fine with  RXJS 5.5.2 in @angular 5.0.3

**Comment by dmugtasimov** [comment]
Well, it is a bug in Angular 5.0.5, because depending on a library with a bug without a proper work-around implemented is a bug. Either Angular should depend on rxjs@5.5.2 (or later version without bug) or a proper work-around must be implemented in Angular itself.

**Comment by vecernik** [comment]
rxjs 5.5.2 works with both angular 5.0.5 and 5.1.0-rc1. rxjs 5.5.3 does not.

**Comment by MickL** [comment]
@dmugtasimov Angular does actually depend on rxjs 5.5.2

**Comment by ocombe** [comment]
The repository itself does, but the packages peer dependency is ^5.5.0

**Comment by SivaPrasanna45** [comment]
I want to use the RxJS5.5.3 , What is the solution for that. Please let me know.

Thanks in advance.

**Comment by vecernik** [comment]
adding pathMatch: 'full' to Route would fix it, but that's not the right way how to fix that imho.

**Comment by ShinDarth** [comment]
I can confirm this issue.

**Comment by kickertw** [comment]
Since route order matters, is best practice to list your routes from most to least specific?  If I switch the order the below, I don't have to have path match in there:

{
path: 'about',
component: AboutComponent
},
{
path: '',
component: HomeComponent
}

**Comment by saicharanmadhu08** [comment]
why does that happen, what does pathMatch: "full" make it to work. What is 5.5.3 missing ?

**Comment by IgorMinar** [comment]
This sounds like an rxjs issue which was already resolved there. Is that correct?

**Comment by scttcper** [comment]
Yes, resolved by downgrading to 5.5.2. People will keep hitting/reporting this error until rxjs makes a new release.

**Comment by wxplovecc** [comment]
@fabiogodoy that works for me

**Comment by bennadel** [comment]
For what it's worth, I just ran into this issue as well; but, none of the affected routes had `path: ""`. I was using a (secondary) named-outlet. All of the routes in the named-outlet were non-empty. But, still, had to revert to RxJS 5.5.2 as per the above guidance.

**Comment by avatsaev** [comment]
How the hell do I even debug something like this?

```
core.es5.js:1020 ERROR Error: Uncaught (in promise): EmptyError: no elements in sequence
EmptyError: no elements in sequence
    at new EmptyError (EmptyError.js:28)
    at FirstSubscriber.webpackJsonp.../../../../rxjs/_esm5/operators/first.js.FirstSubscriber._complete (first.js:154)
    at FirstSubscriber.webpackJsonp.../../../../rxjs/_esm5/Subscriber.js.Subscriber.complete (Subscriber.js:122)
    at MergeMapSubscriber.webpackJsonp.../../../../rxjs/_esm5/operators/mergeMap.js.MergeMapSubscriber._complete (mergeMap.js:150)
    at MergeMapSubscriber.webpackJsonp.../../../../rxjs/_esm5/Subscriber.js.Subscriber.complete (Subscriber.js:122)
    at MapSubscriber.webpackJsonp.../../../../rxjs/_esm5/Subscriber.js.Subscriber._complete (Subscriber.js:140)
    at MapSubscriber.webpackJsonp.../../../../rxjs/_esm5/Subscriber.js.Subscriber.complete (Subscriber.js:122)
    at EmptyObservable.webpackJsonp.../../../../rxjs/_esm5/observable/EmptyObservable.js.EmptyObservable._subscribe (EmptyObservable.js:83)
    at EmptyObservable.webpackJsonp.../../../../rxjs/_esm5/Observable.js.Observable._trySubscribe (Observable.js:172)
    at EmptyObservable.webpackJsonp.../../../../rxjs/_esm5/Observable.js.Observable.subscribe (Observable.js:160)
    at new EmptyError (EmptyError.js:28)
    at FirstSubscriber.webpackJsonp.../../../../rxjs/_esm5/operators/first.js.FirstSubscriber._complete (first.js:154)
    at FirstSubscriber.webpackJsonp.../../../../rxjs/_esm5/Subscriber.js.Subscriber.complete (Subscriber.js:122)
    at MergeMapSubscriber.webpackJsonp.../../../../rxjs/_esm5/operators/mergeMap.js.MergeMapSubscriber._complete (mergeMap.js:150)
    at MergeMapSubscriber.webpackJsonp.../../../../rxjs/_esm5/Subscriber.js.Subscriber.complete (Subscriber.js:122)
    at MapSubscriber.webpackJsonp.../../../../rxjs/_esm5/Subscriber.js.Subscriber._complete (Subscriber.js:140)
    at MapSubscriber.webpackJsonp.../../../../rxjs/_esm5/Subscriber.js.Subscriber.complete (Subscriber.js:122)
    at EmptyObservable.webpackJsonp.../../../../rxjs/_esm5/observable/EmptyObservable.js.EmptyObservable._subscribe (EmptyObservable.js:83)
    at EmptyObservable.webpackJsonp.../../../../rxjs/_esm5/Observable.js.Observable._trySubscribe (Observable.js:172)
    at EmptyObservable.webpackJsonp.../../../../rxjs/_esm5/Observable.js.Observable.subscribe (Observable.js:160)
    at resolvePromise (zone.js:824)
    at resolvePromise (zone.js:795)
    at zone.js:873
    at ZoneDelegate.webpackJsonp.../../../../zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:425)
    at Object.onInvokeTask (core.es5.js:3881)
    at ZoneDelegate.webpackJsonp.../../../../zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:424)
    at Zone.webpackJsonp.../../../../zone.js/dist/zone.js.Zone.runTask (zone.js:192)
    at drainMicroTaskQueue (zone.js:602)
    at ZoneTask.webpackJsonp.../../../../zone.js/dist/zone.js.ZoneTask.invokeTask [as invoke] (zone.js:503)
    at invokeTask (zone.js:1540)
```

**Comment by Cito** [comment]
> How the hell do I even begin debugging something like this?

That's why Angular is called a framework for superheroes. You must be in the zone and then wrap your head around empty promises while carefully observing the first subscriber.

**Comment by bennadel** [comment]
@avatsaev I tried to enable the "Stop on error" in the Source tab of the Chrome Dev Tools. The stack-trace, at least for me, was like 100 calls deep, 80% of which were calls to Observers and Subscribers. At that point, I just Googled for the error message.

**Comment by kylecordes** [comment]
Another success report here. Same error, and the `pathMatch: "full"` work around it. It seems to me this is an unfortunate regression somewhere hiding in RxJS 5.5.3, I would guess it will be found and fixed the next version, though I have not yet looked at the tracker over there to see if it is on the radar.

@avatsaev Asks a great question, and I hear that question **often** from people entering the Angular community. So far the most effective (though inconvenient) answer is that once you been using this entire motley front-end stack a while (not just angular), you get a sense of what kinds of questions to search where... search the Internet, search the issue tracker for libraries you're using directly or indirectly that might be causing your problem, ask someone else who is been down the road before, etc.  And of course you get a sense of what portion of what error message to search for.

**Comment by agrygoriev** [comment]
Same situation with 5.5.2.
`pathMatch: 'full'` resolves the problem.

**Comment by wardbell** [comment]
As a framework developer/maintainer, I really do understand this instinct comment from above.

>Hello, I'm closing this since it is a bug in RXJS and not Angular.

But (as others have noted and I'm just piling on), the comment is insensitive to the Angular developer's predicament. This really is an Angular problem because the developer doesn't have a choice in the matter. Angular has a hard dependency on RxJS ... as it does on zone.js, which, strictly speaking, isn't part of Angular either.

When you know your community will fail because of a hard dependency on another library, it's _your_ problem.  I know we all feel that way ... especially the person who, in a moment of weakness, considered closing this issue.

**Comment by vinothsethupathy007** [comment]
Change  "rxjs": "5.5.2" from "rxjs": "^5.5.2" in package.json file and reinstall packages again
 *npm install*

**Comment by sarunint** [comment]
Just my curiosity, is this issue an Angular's issue or an RxJS issue. I still don't catch here. I read on ReactiveX/rxjs#3152 that it's a problem with Zone.js and waiting for angular/zone.js#935 to be merged.

Well, it's quite complicate. Can anyone clarify this to me?

**Comment by unaor** [comment]
In my case the problem was have a path with children, but none of the children had an empty route...
(i didnt want to have a default component displayed) ,when i left one children with empty path the problem resolved

**Comment by scttcper** [comment]
5.5.4 is out and working for me

**Comment by jasonaden** [comment]
Yes, this was fixed with the new publish of Rx 5.5.4.

**Comment by SofienAbdeddaim** [comment]
@fabiogodoy  work  well, Thanks
